File: BACKEND/app/schemas/checkbox.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CheckboxCreate(BaseModel):
    course_id: int
    video_id: int
    checkbox: int = Field(ge=0)


class CheckboxPublic(CheckboxCreate):
    id: int
    deleted_at: datetime | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)

File: FRONTEND/src/types/checkbox.ts
export type CheckboxCreate = {
  course_id: number;
  video_id: number;
  checkbox: number;
};

export type Checkbox = CheckboxCreate & {
  id: number;
  deleted_at: string | null;
  created_at: string | null;
  updated_at: string | null;
};
